### PR TITLE
Update orderby Dropdown to new layout in web

### PR DESCRIPTION
**Pull request by JuanAlmeida12**
#### What is the purpose of this pull request?

As title says.

#### How should this be manually tested?

[Access this workspace](https://neworderby--storecomponents.myvtex.com)

#### Screenshots or example usage
After:
neworderby--storecomponents myvtex com_accessories
Before:
fixgrid--storecomponents myvtex com_

#### Types of changes

* [ ] Bug fix (a non-breaking change which fixes an issue)
* [ ] New feature (a non-breaking change which adds functionality)
* [x] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/OrderBy.js [hunk 1]
```diff
@@ -64,36 +65,19 @@ class OrderBy extends Component {
   }
 
   render() {
-    const { orderBy, getLinkProps, runtime: { hints: { mobile } }, runtime } = this.props
```
Comment (matheusps) [inline react/OrderBy.js]: This Component does not hold state. You could change it to a functional component and use the `useRuntime` hook.

File: react/searchResult.css [hunk 2]
```diff
@@ -114,8 +114,9 @@
   padding-bottom: 9rem;
 }
 
-.dropdownMobile {
+.dropdownSort {
   margin-top: -2px;
```
Comment (matheusps) [inline react/searchResult.css]: -0.125rem?

File: react/components/SelectionListOrderBy.js [hunk 3]
```diff
@@ -24,6 +24,7 @@ class SelectionListOrderBy extends Component {
   }
 
   static propTypes = {
+    mobile: PropTypes.bool,
```
Comment (matheusps) [inline react/components/SelectionListOrderBy.js]: You could get this value from runtime. This component could also be functional, using the useState hook, like this: `const [showDropdown, setShowDropdown] = useState(false)`

File: react/components/SelectionListOrderBy.js [hunk 4]
```diff
@@ -54,42 +45,57 @@ class SelectionListOrderBy extends Component {
...
-      </div>
-    )
-  }
+      </OutsideClickHandler>
+    </div>
+
+  )
+}
+
+SelectionListOrderBy.propTypes = {
```
Comment (matheusps) [inline react/components/SelectionListOrderBy.js]: docs ?

File: react/OrderBy.js [hunk 5]
```diff
@@ -40,64 +39,45 @@ export const SORT_OPTIONS = [
...
-    }).isRequired,
-  }
-
-  get sortingOptions() {
+const OrderBy = ({
+  orderBy,
+  getLinkProps,
+  intl,
+}) => {
+  const sortingOptions = () => {
```
Comment (lucasecdb) [inline react/OrderBy.js]: what do you think about refactoring this to [`useMemo`](https://reactjs.org/docs/hooks-reference.html#usememo)?

File: react/components/SelectionListOrderBy.js [hunk 6]
```diff
@@ -1,43 +1,34 @@
...
+const SelectionListOrderBy = ({
+  orderBy,
+  getLinkProps,
+  options,
+  intl,
+}) => {
+  const [showDropdown, setShowDropdown] = useState(false)
 
-  handleDropdownBtClick = e => {
+  const handleDropdownBtClick = e => {
```
Comment (lucasecdb) [inline react/components/SelectionListOrderBy.js]: add [`useCallback`](https://reactjs.org/docs/hooks-reference.html#usecallback)

File: react/components/SelectionListOrderBy.js [hunk 7]
```diff
@@ -1,43 +1,34 @@
...
-    getLinkProps: PropTypes.func,
-    options: PropTypes.arrayOf(PropTypes.shape({
-      label: PropTypes.string,
-      value: PropTypes.string,
-    })),
-    intl: intlShape,
-    runtime: PropTypes.shape({
-      navigate: PropTypes.func,
-    }),
+  const handleOutsideClick = e => {
```
Comment (lucasecdb) [inline react/components/SelectionListOrderBy.js]: `useCallback`

File: react/components/SelectionListOrderBy.js [hunk 8]
```diff
@@ -54,42 +45,63 @@ class SelectionListOrderBy extends Component {
     })
   }
 
-  getOptionTitle = option => {
-    const { options, intl } = this.props
+  const getOptionTitle = option => {
```
Comment (lucasecdb) [inline react/components/SelectionListOrderBy.js]: `useCallback`

File: react/components/SelectionListOrderBy.js [hunk 9]
```diff
@@ -18,19 +18,18 @@ const SelectionListOrderBy = ({
...
-  const handleDropdownBtClick = e => {
-    e.preventDefault()
+  const handleDropdownBtClick = useCallback(() => {
     setShowDropdown(!showDropdown)
-  }
+  }, [])
 
-  const handleOutsideClick = e => {
-    e.preventDefault()
+  const handleOutsideClick = useCallback(() => {
```
Comment (matheusps) [inline react/components/SelectionListOrderBy.js]: You can simply `const handleOutsideClick = useCallback( () => setShowDropdown(false) )

File: react/components/SelectionListOrderBy.js [hunk 10]
```diff
@@ -45,9 +44,9 @@ const SelectionListOrderBy = ({
     })
   }
 
-  const getOptionTitle = option => {
+  const getOptionTitle = useCallback((option) => {
```
Comment (matheusps) [inline react/components/SelectionListOrderBy.js]: This works ? 🤔Shouldn't be something like `const getOptionTitle = useCallback(() => find(propEq('value', option), options).label, [option])`  ?

Comment (matheusps) [reply react/components/SelectionListOrderBy.js]: @lucasecdb, what do u think?

Comment (lucasecdb) [reply react/components/SelectionListOrderBy.js]: `option` is actually a parameter, we only need to add `options` to the deps array

Comment (lucasecdb) [reply react/components/SelectionListOrderBy.js]: you can see the usage [here](https://github.com/vtex-apps/search-result/pull/116/files#diff-01b05f9607839211d3b9f685410dc260R75)

File: react/components/SelectionListOrderBy.js [hunk 11]
```diff
@@ -1,43 +1,28 @@
...
+  const handleDropdownBtClick = useCallback(
+    () => setShowDropdown(!showDropdown),
+    [showDropdown]
+  )
 
-  handleOutsideClick = e => {
-    this.setState({ showDropdown: false })
-  }
+  const handleOutsideClick = useCallback(() => setShowDropdown(false), [
+    showDropdown,
```
Comment (lucasecdb) [inline react/components/SelectionListOrderBy.js]: you don't need `showDropdown` here. please, read the [`useCallback` docs](https://reactjs.org/docs/hooks-reference.html#usecallback)

**Pull request closed without merging**